tsconfig.json

```diff
@@ -6,7 +6,8 @@
     "strict": true,
     "declaration": false,
     "declarationMap": false,
-    "allowJs": true,
+    "allowJs": false,
+    "noImplicitAny": false,
```

Comment:
@HazAT @lynnagara There are issues with importing React components from `*.jsx` files, and I don't think redeclaring them as `const FooBarAny: any = FooBar;` is the right approach to migrate Sentry to TypeScript in the long-term.

Since this file is extending `@sentry/typescript/tsconfig.json`  (https://github.com/getsentry/sentry-javascript/blob/5d670a7781b6e2144809c002e9332e2cc9dd239c/packages/typescript/tsconfig.json), I'm proposing to provide the following overrides that are specific for this project: 

```
    "allowJs": false,
    "noImplicitAny": false,
```

The behavioural changes are as follows: 

- Setting `allowJs` to be `false` prevents the TypeScript engine from processing `*.jsx` files and as well as inferring types of imported js code. This resolves cryptic proptype type issues for React components sourced in `*.jsx` files . 

- The implications of disabling `allowJs` means that imported modules from `*.jsx` files are implied to have an `any` type. Hence, we need `noImplicitAny` to be `false`.